src/Stratis.Bitcoin/Consensus/ConsensusManager.cs

```diff
@@ -190,9 +193,13 @@ public async Task InitializeAsync(ChainedHeader chainTip)
                 pendingTip = chainTip.FindAncestorOrSelf(consensusTipHash);
 
                 if ((pendingTip != null) && (this.chainState.BlockStoreTip.Height >= pendingTip.Height))
+                {
+                    this.StatusMessage = null;
                     break;
+                }
 
                 this.logger.LogInformation("Consensus at height {0} is ahead of the block store at height {1}, rewinding consensus.", pendingTip, this.chainState.BlockStoreTip);
+                this.StatusMessage = string.Join("Rewinding consensus from height {0} to {1}.", pendingTip.Height, this.chainState.BlockStoreTip.Height);
```

Comment:
I'd rather have a method that returns the state of the ConsensusManager, and the nodeController call this method and decide what to do with it.
The state doesn't have to be an enum but can be an object containing the tip, the rewinding or not status etc..

Reply:
@bokobza see my comments here #3021 .. can you comment on my last comment there?